csp_billing_adapter_microsoft/plugin.py

```diff
@@ -81,55 +77,54 @@ def get_account_info(config: Config):
     The information contains the metadata for compute and network.
     """
     account_info = _get_metadata()
-    account_info['compute'] = json.loads(account_info.get('compute', '{}'))
-    account_info['network'] = json.loads(account_info.get('network', '{}'))
     account_info['cloud_provider'] = get_csp_name(config)
 
     return account_info
 
 
-def get_api_header():
-    """
-    Get the header to be used in requests
-
-    Prefer IMDS which requires a token.
-    """
-    request = urllib.request.Request(
-        METADATA_TOKEN_URL,
-        headers={'Metadata': 'True'},
-        method='GET'
-    )
-
+def _get_metadata():
+    metadata = {}
     try:
-        token = urllib.request.urlopen(request).read().decode()
-    except urllib.error.URLError as error:
-        log.error(f'Failed to retrieve metadata token: {str(error)}')
-        return {}
+        metadata = _get_signature()
+        metadata['compute'] = _get_compute_metadata()
+    except ValueError as error:
+        log.error('Could not load JSON from metadata: {}'.format(error))
+        for key in ['compute', 'signature']:
+            if not metadata.get(key):
+                metadata[key] = {}
 
-    return token
+    return metadata
 
 
-def _get_metadata():
-    metadata_options = ['compute', 'network']
+def _get_compute_metadata():
+    compute = json.loads(_fetch_metadata(COMPUTE_URL))
+
     return {
```

Comment:
This is only returning offer and location but I might need subscriptionID and resourceId which is also available. Can we add this in as well?

Reply:
we have `subscriptionId` available in the [signature](https://github.com/SUSE-Enceladus/csp-billing-adapter-microsoft/pull/4#issuecomment-1563159461)

I can add `resourceId`, no problem